cpp/src/strings/convert/convert_urls.cu

```diff
@@ -325,85 +378,60 @@ std::unique_ptr<column> url_decode(
   rmm::cuda_stream_view stream,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource())
 {
+  constexpr int threadblock_size = 128;
+  constexpr int block_size       = 256;
+  constexpr int num_threadblocks = 65536;
```

Comment:
I wonder how these hard-coded values are correct and can be applied to various architectures. Should we use some functions to compute them instead?

Reply:
These hard-coded values should not impact correctness. From performance's point of view, I have experimented with different `block_size` and performance impact is minimal.

I agree setting `num_threadblocks` to a constant value is wasteful if we do not have a lot of strings. Switch to use number of warps = number of strings in https://github.com/rapidsai/cudf/pull/8622/commits/f4785201975ef154dfd8d88fe21a04dd9bdde96d.